game/scripts/vscripts/abilities/oaa_zuus_cloud.lua

```diff
@@ -0,0 +1,173 @@
+  end
+end
+
+function modifier_zuus_cloud_oaa:OnDestroy()
+  if IsServer() then
+    ParticleManager:DestroyParticle(self.cloud_particle, false)
+    ParticleManager:ReleaseParticleIndex(self.cloud_particle)
+    self.cloud_particle = nil
+  end
+end
+
+function modifier_zuus_cloud_oaa:DeclareFunctions()
+	local funcs =
+	{
+    MODIFIER_PROPERTY_ABSOLUTE_NO_DAMAGE_PHYSICAL,
+    MODIFIER_PROPERTY_ABSOLUTE_NO_DAMAGE_MAGICAL,
+    MODIFIER_PROPERTY_ABSOLUTE_NO_DAMAGE_PURE,
+    MODIFIER_EVENT_ON_ATTACKED
+	}
+	return funcs
+end
+
+function modifier_zuus_cloud_oaa:GetAbsoluteNoDamagePhysical()
+  return 1
+end
+
+function modifier_zuus_cloud_oaa:GetAbsoluteNoDamageMagical()
+  return 1
+end
+
+function modifier_zuus_cloud_oaa:GetAbsoluteNoDamagePure()
+  return 1
+end
+
+function modifier_zuus_cloud_oaa:OnAttacked(params)
+  local parent = self:GetParent()
+
+  local attacker = params.attacker;
+
+  if attacker:IsHero() and params.ranged_attack then
+    return parent:SetHealth(parent:GetHealth() - 2)
+  elseif attacker:IsHero() and not params.ranged_attack then
+    return parent:SetHealth(parent:GetHealth() - 4)
+  else
+    return parent:SetHealth(parent:GetHealth() - 1)
+  end
+end
+
+function modifier_zuus_cloud_oaa:OnIntervalThink()
+  if self.LastStrike == nil or GameRules:GetDOTATime(false, false) - self.LastStrike > self.Interval then
+    local caster = self:GetCaster()
+    local targets = FindUnitsInRadius(
+      caster:GetTeamNumber(),
+      self:GetParent():GetAbsOrigin(),
+      nil,
+      self.Radius,
+      DOTA_UNIT_TARGET_TEAM_ENEMY,
+      DOTA_UNIT_TARGET_HERO,
+      DOTA_UNIT_TARGET_FLAG_NONE,
+      FIND_CLOSEST,
+      false
+    )
+    if #targets == 0 then
+      targets = FindUnitsInRadius(
+        caster:GetTeamNumber(),
+        self:GetParent():GetAbsOrigin(),
+        nil,
+        self.Radius,
+        DOTA_UNIT_TARGET_TEAM_ENEMY,
+        DOTA_UNIT_TARGET_CREEP,
+        DOTA_UNIT_TARGET_FLAG_NONE,
+        FIND_CLOSEST,
+        false
+      )
+    end
+
+    if #targets > 0 then
+      self:CastLightningBolt(targets[1])
+      self.LastStrike = GameRules:GetDOTATime(false, false)
+    end
+  end
+end
+
+function modifier_zuus_cloud_oaa:CastLightningBolt(target)
+  local caster = self:GetCaster()
+  local parent = self:GetParent()
+  local lightning_bolt_ability = caster:FindAbilityByName( 'zuus_lightning_bolt' )
+  local sight_radius =  0
+  if GameRules:IsDaytime() then
+    sight_radius = lightning_bolt_ability:GetSpecialValueFor("sight_radius_day")
+  else
+    sight_radius = lightning_bolt_ability:GetSpecialValueFor("sight_radius_night")
+  end
+  local sight_duration = lightning_bolt_ability:GetSpecialValueFor("sight_duration")
+
+  if lightning_bolt_ability:GetLevel() > 0 then
+
+    CreateModifierThinker( caster, nil, "modifier_true_sight_oaa_thinker", {duration = sight_duration, radius = sight_radius }, target:GetAbsOrigin(), caster:GetTeamNumber(), false )
```

Comment:
Could try reusing the vanilla `modifier_zuus_lightningbolt_vision_thinker`.

Reply:
I was not aware of that! nice!
Can doit right now, but I'll add it for sure!

Reply:
turns out there is a modifier_truesight, that is what vanila uses.
modifier_zuus_lightningbolt_vision_thinker gives vision, I I could not make it work (probably have to figure out how to pass the range on the constructor and linking to zuus W does not work.